ipaclient/install/client.py

```diff
@@ -1654,13 +1670,62 @@ def client_dns(server, hostname, options):
                        hostname, ex)
         dns_ok = False
 
+    # Setup DNS over TLS
+    if options.dns_over_tls:
+        # setup and enable Unbound as resolver
+        server_ip = str(list(dnsutil.resolve_ip_addresses(server))[0])
+        forward_addr = "forward-addr: %s#%s" % (server_ip, server)
+        ipautil.copy_template_file(
+            paths.UNBOUND_CONF_SRC,
+            paths.UNBOUND_CONF,
+            dict(
+                TLS_CERT_BUNDLE_PATH=os.path.join(
+                    paths.OPENSSL_CERTS_DIR, "ca-bundle.crt"),
+                FORWARD_ADDRS=forward_addr
+            )
+        )
+        sr = services.knownservices["systemd-resolved"]
+        if sr.is_running():
+            sr.stop()
+            sr.disable()
+
+        nm = services.knownservices["NetworkManager"]
+        if nm.is_enabled():
+            with open(paths.NETWORK_MANAGER_IPA_CONF, "w") as f:
+                dns_none = [
+                    "# auto-generated by IPA installer",
+                    "[main]",
+                    "dns=none\n"
+                ]
+                f.write("\n".join(dns_none))
+            nm.reload_or_restart()
+
+        # Overwrite resolv.conf to point to Unbound
+        cfg = [
+            "# auto-generated by IPA installer",
+            "search .",
+            "nameserver 127.0.0.55\n"
+        ]
```

Comment:
both `dns_none` and `cfg` configurations are static and basically can be put to templates in the `client/share/` space, then can be read from there. 

This PR right also uses two different methods to present in-code templates (`textwrap.dedent()` and `"\n".join([..])`). Perhaps it is better to adjust to a single approach and use templates too?